tw-tasks-core/src/main/java/com/transferwise/tasks/TasksService.java

```diff
@@ -58,6 +61,8 @@ public class TasksService implements ITasksService, GracefulShutdownStrategy, In
   private IEnvironmentValidator environmentValidator;
   @Autowired
   private ICoreMetricsTemplate coreMetricsTemplate;
+  @Autowired(required = false)
+  private List<ITaskRegistrationDecorator> taskRegistrationInterceptors = new ArrayList<>();
```

Comment:
Maybe taskRegistrationInterceptors > taskRegistrationDecorator

Reply:
Let's keep decorator, usually decorator gives expectation that you are modifying something
interceptor not really it means you just react